### PR TITLE
Add test module Audit: aureport.pm

### PR DESCRIPTION
**Pull request by ShawnHa0**
Verify the "aureport" utility can generate default audit report and reports with specifc parameters

- Related ticket: https://progress.opensuse.org/issues/81772
- Needles: N/A
- Verification run: 
  - o3: https://openqa.opensuse.org/tests/2201474
  - osd: https://openqa.suse.de/tests/8206234

### CODE REVIEW

**Review by rfan1 — APPROVED** [review]
lgtm

**Review by Amrysliu — APPROVED** [review]

**Comment by lilyeyes** [comment]
https://openqa.opensuse.org/tests/2198454 is wrong.

File: tests/security/audit/aureport.pm [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-Later
+#
+# Summary: Verify the "aureport" utility can generate default audit report and reports with specifc parameters
+# Maintainer: llzhao <llzhao@suse.com>, shawnhao <weixuan.hao@suse.com>
+# Tags: poo#81772
```
Comment (lilyeyes) [inline tests/security/audit/aureport.pm:6]: Please add the tc#$testopia-test-num here also.

**Comment by ShawnHa0** [comment]
> https://openqa.opensuse.org/tests/2198454 is wrong.

Updated with the right one now.

**Review by jouyingbin — APPROVED** [review]
LGTM

File: tests/security/audit/aureport.pm [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    assert_script_run('systemctl start apparmor');
+
+    # Stop auditd before checking records in case the following executions will generate new audit records
+    assert_script_run('systemctl stop auditd');
+
+    # Generate report by default
+    assert_script_run('aureport');
+
+    # Generate AVC report
+    validate_script_output('aureport -a', sub { m/AVC/ });
```
Comment (lilyeyes) [inline tests/security/audit/aureport.pm:34]: It is optional:
Could you please compare the values of $num1 and $num2 to check if they are the same?
$numb1 is the "Number of AVC's: $number" in the outputs of "assert_script_run('aureport');"
$numb2 is the total number for the records in 'aureport -a'
See https://bugzilla.suse.com/tr_show_case.cgi?case_id=1768552 for the outputs examples.

Comment (lilyeyes) [reply tests/security/audit/aureport.pm:34]: `aureport - a tool that produces summary reports of audit daemon logs` so adding this test step can verify the summary is correct.

Comment (ShawnHa0) [reply tests/security/audit/aureport.pm:34]: Sure, I will add the comparison

Comment (ShawnHa0) [reply tests/security/audit/aureport.pm:34]: AVC report counts checking added now.

File: tests/security/audit/aureport.pm [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    my $tmp_output = '/tmp/output';
+    my $tmp_backup = '/tmp/backup';
+    assert_script_run("aureport -a > $tmp_output");
+    assert_script_run("tail -1 $tmp_output > $tmp_backup");
+    my $avc_num = script_output("cat $tmp_backup | cut -d '.' -f 1");
+    if ($default_num == $avc_num) {
+        record_info('Comparison passed: ', 'AVC reports counted in default report is the same as generated by AVC reports');
+    } else {
+        record_info('Error: ', 'AVC reports counted in default report do not match the counts in AVC reports');
+    }
```
Comment (lilyeyes) [inline tests/security/audit/aureport.pm:47]: it should be failed here

Comment (ShawnHa0) [reply tests/security/audit/aureport.pm:47]: Result 'fail' added along with tc tags for previous cases.

**Review by lilyeyes — APPROVED** [review]
LGTM.

**Pull request merged**